### PR TITLE
chore: E2E Dashboards Cross-references

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Adds E2E tests for cross-referenced Dashboards in Charts list and Explore. Filtering was already tested in the [original PR.](https://github.com/apache/superset/pull/21760) 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
All tests should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/cypress-base/cypress/integration/chart_list/list.test.ts [hunk 1]
```diff
@@ -43,14 +47,70 @@ function confirmDelete() {
...
+    beforeEach(() => {
+      visitChartList();
+    });
+
+    before(() => {
+      cy.createSampleDashboards([0, 1, 2, 3]);
+      cy.createSampleCharts([0]);
+    });
+
+    it('should the cross-referenced dashboards in the table cell', () => {
```
Comment (michael-s-molina) [inline superset-frontend/cypress-base/cypress/integration/chart_list/list.test.ts:71]:
```suggestion
    it('shows the cross-referenced dashboards in the table cell', () => {
```

File: superset-frontend/cypress-base/cypress/integration/chart_list/list.test.ts [hunk 2]
```diff
@@ -43,14 +47,70 @@ function confirmDelete() {
...
+        .eq(1)
+        .find('[data-test="table-row-cell"]')
+        .find('[data-test="crosslinks"]')
+        .contains('Tabbed Dashboard')
+        .invoke('removeAttr', 'target')
+        .click();
+      cy.wait('@get');
+    });
+
+    it('should the newly added dashboards in a tooltip', () => {
```
Comment (michael-s-molina) [inline superset-frontend/cypress-base/cypress/integration/chart_list/list.test.ts:93]:
```suggestion
    it('shows the newly added dashboards in a tooltip', () => {
```

File: superset-frontend/cypress-base/cypress/integration/explore/chart.test.js [hunk 3]
```diff
@@ -16,7 +16,113 @@
...
+    cy.visit(CHART_LIST);
+    cy.wait('@filtering');
+  });
+
+  before(() => {
+    cy.createSampleDashboards(SAMPLE_DASHBOARDS_INDEXES);
+    cy.createSampleCharts([0]);
+  });
+
+  it('Shows the cross referenced dashboards', () => {
```
Comment (michael-s-molina) [inline superset-frontend/cypress-base/cypress/integration/explore/chart.test.js:103]:
```suggestion
  it('shows the cross-referenced dashboards', () => {
```

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the tests @geido!

**Review by jinghua-qa — APPROVED** [review]
LGTM

**Pull request merged**